pkg/cloudprovider/aws/fleet/capacity.go

```diff
@@ -18,73 +18,75 @@ import (
 	"fmt"
 
-	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
 	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	v1 "k8s.io/api/core/v1"
 )
 
 // Capacity cloud provider implementation using AWS Fleet.
 type Capacity struct {
-	spec                   *v1alpha1.ProvisionerSpec
-	launchTemplateProvider *LaunchTemplateProvider
-	subnetProvider         *SubnetProvider
-	nodeFactory            *NodeFactory
-	instanceProvider       *InstanceProvider
+	spec             *v1alpha1.ProvisionerSpec
+	nodeFactory      *NodeFactory
+	packing          cloudprovider.Packer
+	instanceProvider *InstanceProvider
+	vpc              *VPCProvider
 }
 
 // Create a set of nodes given the constraints.
-func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.CapacityConstraints) (cloudprovider.CapacityPacking, error) {
-	// 1. Compute Constraints
-	launchTemplate, err := c.launchTemplateProvider.Get(ctx, c.spec.Cluster)
+func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constraints) (cloudprovider.PodPackings, error) {
+	// 1. Compute Packing given the constraints
+	packings, err := c.packing.Pack(ctx, constraints.Pods)
 	if err != nil {
-		return nil, fmt.Errorf("getting launch template, %w", err)
+		return nil, fmt.Errorf("computing bin packing, %w", err)
 	}
-	instancePackings, err := c.instanceProvider.GetPackings(ctx, constraints.Pods, constraints.Overhead)
+
+	launchTemplate, err := c.vpc.GetLaunchTemplate(ctx, c.spec.Cluster)
 	if err != nil {
-		return nil, fmt.Errorf("computing bin packing, %w", err)
+		return nil, fmt.Errorf("getting launch template, %w", err)
 	}
-	zonalSubnets, err := c.getConstrainedZonalSubnets(ctx, constraints)
+
+	zonalSubnetOptions, err := c.vpc.GetZonalSubnets(ctx, constraints, c.spec.Cluster.Name)
 	if err != nil {
 		return nil, fmt.Errorf("getting zonal subnets, %w", err)
 	}
 
 	// 2. Create Instances
 	var instanceIds []*string
-	for _, instancePacking := range instancePackings {
-		instanceId, err := c.instanceProvider.Create(ctx, launchTemplate, instancePacking.InstanceTypeOptions, zonalSubnets)
+	podsMapped := make(map[string][]*v1.Pod)
+	for _, packing := range packings {
+		ec2InstanceID, err := c.instanceProvider.Create(ctx, packing.InstanceTypeOptions, launchTemplate, zonalSubnetOptions)
 		if err != nil {
 			// TODO Aggregate errors and continue
 			return nil, fmt.Errorf("creating capacity %w", err)
 		}
-		instanceIds = append(instanceIds, instanceId)
+		podsMapped[*ec2InstanceID] = packing.Pods
+		instanceIds = append(instanceIds, ec2InstanceID)
 	}
 
 	// 3. Convert to Nodes
 	nodes, err := c.nodeFactory.For(ctx, instanceIds)
 	if err != nil {
 		return nil, fmt.Errorf("determining nodes, %w", err)
 	}
-
-	// 4. Construct capacity packing
-	capacityPacking := cloudprovider.CapacityPacking{}
-	for i, node := range nodes {
-		capacityPacking[node] = instancePackings[i].Pods
+	nodePackings := make(cloudprovider.PodPackings)
```

Comment:
This naming seems odd to me. I typically try to just name my variables according to the type. In this case, it's not clear if the concept we're dealing with here is a "pod packing" or a "node packing". Whatever we call it, let's be consistent. 

I also see that you're a little cornered since you have two packing concepts in this function (not to mention the misnamed "podpackings"). Off the top of my head, I might call the one above an "InstancePacking" since it's EC2 Instance specific and the second one a "NodePacking" since its a kubernetes node.